test/doc/websocket_2_handshaking.cpp

```diff
@@ -132,6 +132,97 @@ snippets()
 
     //]
     }
+
+    {
+        //[code_websocket_2_6
+
+        // a function to select the most preferred protocol from a comma-separated list
+        auto best_protocol = [](string_view offered_tokens) -> std::string
+        {
+            // tokenize the Sec-Websocket-Protocol header offered by the client
+            http::token_list offered( offered_tokens );
+
+            // an array of protocols supported by this server
+            // in descending order of preference
+            static const std::array<string_view, 3>
+                supported = {
+                "v3.my.chat",
+                "v2.my.chat",
+                "v1.my.chat"
+            };
+
+            std::string result;
+
+            for (auto proto : supported)
+            {
+                auto iter = std::find(offered.begin(), offered.end(), proto);
+                if (iter != offered.end())
+                {
+                    // we found a supported protocol in the list offered by the client
+                    result.assign(proto.begin(), proto.end());
+                    break;
+                }
+            }
+
+            return result;
+        };
+
+
+        // This buffer is required for reading HTTP messages
+        flat_buffer buffer;
+
+        // Read the HTTP request ourselves
+        http::request<http::string_body> req;
+        http::read(sock, buffer, req);
+
+        // See if it's a WebSocket upgrade request
+        if(websocket::is_upgrade(req))
+        {
+            // we store the selected protocol in a std::string here
+            // because we intend to capture it in the decorator's lambda
+            // below
```

Comment:
widow

Reply:
Perhaps I misunderstood. I put this comment in because I thought you were asking for exposition of why the protocol was being captured in a string rather than a string_view

Reply:
What I mean is that the word "below" is all by iteself

Reply:
Oh I see. I was trying to avoid crossing the undefined column boundary.